### PR TITLE
JDK-8253239: Disable VS warning C4307

### PR DESCRIPTION
**Pull request by MBaesken**
hello, this fixes the build when using VS2017. VS2019 does not have the issue as far as I know.
failure was 

> ./test/hotspot/gtest/utilities/test_align.cpp(96): error C2220: warning treated as error - no 'object' file generated
> ./test/hotspot/gtest/utilities/test_align.cpp(156): note: see reference to function template instantiation 'void static_test_alignments<T,A>(void)' being compiled
> with
> [
> T=int64_t,
> A=uint8_t
> ]
> ./test/hotspot/gtest/utilities/test_align.cpp(162): note: see reference to function template instantiation 'void test_alignments<int64_t,uint8_t>(void)' being compiled
> ./test/hotspot/gtest/utilities/test_align.cpp(96): warning C4307: '+': integral constant overflow
>
>

This might be related to an issue fixed at least in some versions of VS2019 , that is discussed here :

https://developercommunity.visualstudio.com/content/problem/211134/unsigned-integer-overflows-in-constexpr-functionsa.html

https://bugs.openjdk.java.net/browse/JDK-8253239

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253239](https://bugs.openjdk.java.net/browse/JDK-8253239): Disable VS warning C4307

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/237/head:pull/237`
`$ git checkout pull/237`

### CODE REVIEW

File: make/autoconf/flags-cflags.m4
Comment (shipilev) [inline make/autoconf/flags-cflags.m4:138]: Seems like this file uses 2 spaces as indenting, 4 spaces are used here.

Comment (MBaesken) [reply make/autoconf/flags-cflags.m4]: You Are right, I adjusted the indenting to 2 sapces and did a git commit  --amend and repushed it.

**Review by shipilev — COMMENTED** [review]
This looks fine to me, but build maintainers should approve.

File: make/autoconf/flags-cflags.m4
Comment (erikj79) [inline make/autoconf/flags-cflags.m4:137]: I would like a comment here explaining why we need to disable this warning, since it's caused by a faulty behavior in the compiler.

File: make/autoconf/flags-cflags.m4
Comment (erikj79) [inline make/autoconf/flags-cflags.m4:138]:
```suggestion
        # VS2017 incorrectly triggers this warning for constexpr
        DISABLED_WARNINGS+=" 4307"
```

**Comment by erikj79** [comment]
I left a suggestion for a comment. You should be able to just accept it if you are happy with it. If you would rather write something yourself, I recommend committing without --amend. The Skara bot will automatically squash all your changes before merging this to the main repo.

**Review by TheRealMDoerr — APPROVED** [review]

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**
